### ISSUE TITLE
h5 model and about newer model

### ISSUE THREAD

**Issue by Odedbenc** [Ant-Brain/EfficientWord-Net#27]
Hi,
I am playing with the model, and wanted to train with new data on the existing model. Since, I cannot retrieve the model's weights from the tflite model, is there a chance to get the h5 model from you??

**Comment by aman-17** [comment]
Yes, give me a little time, and I will provide it.

**Comment by Odedbenc** [comment]
Hi my friend,
you sent me the tflite model, I was talking about the h5 model...
I want to use it as a transfer learning model, and to fine tune it to my
task..
If it is still available, I will be very grateful.
thanks, Oded

**Comment by Odedbenc** [comment]
By the way,
You did a great job with the model, and the paper is also well-written, and
very interesting.

**Comment by aman-17** [comment]
> Hi my friend, you sent me the tflite model, I was talking about the h5 model... I want to use it as a transfer learning model, and to fine tune it to my task.. If it is still available, I will be very grateful. thanks, Oded
> […](#)

Hey Oded, 
Unfortunately, I didn't find the .h5 base model because this project was developed while I was an undergrad, all the models and checkpoints were saved in the .edu drive. But now, my university has deactivated the account.

**Comment by Odedbenc** [comment]
Ohh, that's a big bummer 😞..
So I can not recreate the keras base model with the tflite one, right?
Thanks for your response

**Comment by aman-17** [comment]
> Ohh, that's a big bummer 😞.. So I can not recreate the keras base model with the tflite one, right? Thanks for your response
> […](#)

You can't. But we are working on developing a new version of this project. We will publish the .h5 file. If you want to recreate this model, I can share the dataset.

**Comment by Odedbenc** [comment]
I will really appreciate it. If you can send me a we transfer or a link to
S3 bucket that would be great!!

**Comment by aman-17** [comment]
Here are the links: https://github.com/Ant-Brain/wakeword_dataset_generator, https://drive.google.com/file/d/1f6dp72D9WxErXvaZP6KIBLv4-eKpvLZa/view?usp=sharing

**Comment by Odedbenc** [comment]
Thank you very much, I highly appreciate it 🙏

**Comment by Odedbenc** [comment]
Hi again, can you please send me the ligmelcalc.tflite converter?

**Comment by Odedbenc** [comment]
Sorry. I found it.

**Comment by TheSeriousProgrammer** [comment]
We have trained a newer model lately , kindly test it out, just tiding up the train flow in that , soon training code too will be released

**Comment by Odedbenc** [comment]
Nice, really appreciate it.
How do you synthesize the data?

**Comment by TheSeriousProgrammer** [comment]
The newer model has been trained with a dataset derived from People speech dataset from MLCommons, the original dataset has a lot of imbalances and they havent pre split into train and test, so we had to create a more polished derivative of it , will share them soon . For now you can now have a look at the uncleaned training code if you want 

https://github.com/Ant-Brain/EfficientWord-Net-Trainer.git

simple_trainer.py is the entry  point

**Comment by Odedbenc** [comment]
I also trained a model using mlcommon, I wonder what kind of metric you
used to eliminate similar words.. I used a combination of jaccard
similarity and levinstein distance..

**Comment by Odedbenc** [comment]
And by the way, I played with some other versions of efficient net, and
they gave better results.

**Comment by TheSeriousProgrammer** [comment]
In this derivative we didnt eliminate the words. if we keep close enough words, the model tends to learn better
Even in this newer model struggles to differentiate words like "lights on" and "lights off", but keeping them in the dataset has certainly increased the discriminative powers of the model

Moreover we have used ArcLoss instead of triplet loss, (currently state of the art face rec papers are using the same), that too has played quite a role

**Comment by Odedbenc** [comment]
Interesting. I would really like to have a look at it. By the way, the link
you sent me is broken.. 404..

**Comment by Odedbenc** [comment]
Sorry for all the questions, but I really want to understand more about the
mfcc conversation you did, tflite is not working with a more simple tf
functions to create mfcc?
Is it possible to connect the mfcc conversation to the model as a layer in
the network and to wrap it up to one tflite model?

**Comment by TheSeriousProgrammer** [comment]
The repo was private , can you check again?
Moreover the logmelcalc.tflite still works, can you check again? Cause I did some tests on the older model too to check if the api changes where working

The problem with adding logmelcalc.tflite to the model itself is that, the converted was generator has a very tiny bug which makes it crash on adding the batch dimension, its working in the inference pipeline as there was no need to make batch inferences yet

The newer model doesnt make use of the same anymore, instead on pure numpy operators , moreover the new preprocessor can't be added to the model itself as, some the operators involved can't be exported to onnx

**Comment by TheSeriousProgrammer** [comment]
(about the preprocess being a part of the model itself)
I am not able to dive too much into it due to time contraints, but it should be theoretically possible
Kindly share us if you make any progress in that front : )

**Comment by Odedbenc** [comment]
Of course, thank you very much for your sharing. ,🙏
